extensions/binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java

```diff
@@ -166,15 +173,27 @@ private synchronized boolean updateWeatherData() {
         return false;
     }
 
+    private final Cache<String, String> CACHE = CacheBuilder.newBuilder().expireAfterWrite(10, TimeUnit.SECONDS)
+            .maximumSize(5).build(new CacheLoader<String, String>() {
+                @Override
+                public String load(String query) throws Exception {
+                    try {
+                        URL url = new URL("https://query.yahooapis.com/v1/public/yql?format=json&q="
+                                + query.replaceAll(" ", "%20").replaceAll("'", "%27"));
+                        URLConnection connection = url.openConnection();
+                        return IOUtils.toString(connection.getInputStream());
+                    } catch (MalformedURLException e) {
+                        logger.debug("Constructed query url '{}' is not valid: {}", query, e.getMessage());
+                        return null;
```

Comment:
Just notices that they don't like `null` here, so this sucks.

Hang on a second...